plugins/rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -262,34 +262,50 @@ def _setup_endpoint(self, region_string: str) -> str:
 
         return endpoint
 
-    def get_agents_by_ip(self, ip_address: str) -> List[Dict[str, Any]]:
+    def get_agents_by_ip(self, ip_address: str, next_cursor: str = None) -> List[Dict[str, Any]]:
+        start_time = datetime.now()
         agents = []
         payload = {
             "query": "query( $orgId:String! ) { organization(id: $orgId) { assets( first: 10000 ) { pageInfo { hasNextPage endCursor } edges { node { id platform host { vendor version description hostNames { name } primaryAddress { ip mac } uniqueIdentity { source id } attributes { key value } } publicIpAddress agent { agentSemanticVersion agentStatus quarantineState { currentState } } } } } } }",
             "variables": {"orgId": self.org_key},
         }
 
-        self.logger.info(f"Getting all agents by IP: {ip_address}...")
+        self.logger.info(f"Getting first page of agents by IP: {ip_address}...")
```

Comment:
if the customer runs this action is it not going to still get the first page if even they've specified the next_cursor?

Reply:
You're right. Not tested for subsequent runs. I've updated this now to skip first page on subsequent runs.